.github/workflows/main.yml

```diff
@@ -0,0 +1,97 @@
+  HELM_REPO: Alfresco/charts
+  HELM_REPO_BRANCH: master
+  HELM_REPO_SUBFOLDER: incubator
+
+jobs:
+
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-helm-docs@v1.9.2
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-kubepug@v1.9.2
+      - uses: Alfresco/alfresco-build-tools/.github/actions/pre-commit@v1.9.2
+      - uses: bridgecrewio/checkov-action@v12.1708.0
+        with:
+          framework: kubernetes
+
+  build:
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    env:
+      TARGET_CHARTS_PATH: charts
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Define test namespace
+        id: define-namespace
+        env:
+          RUN_ID: ${{ github.run_id }}
+          REPO_NAME: ${{ github.event.repository.name }}
+        run: echo "::set-output name=name::gh-$REPO_NAME-$RUN_ID"
+
+      - name: Build
+        uses: Alfresco/alfresco-build-tools/.github/actions/helm-build-chart@AAE-10136-review-helm-actions
+        with:
+          chart-dir: ${{ env.CHART_DIR }}
+
+      - name: Unit Tests
+        uses: Alfresco/alfresco-build-tools/.github/actions/helm-unit-tests@AAE-10136-review-helm-actions
+        with:
+          chart-dir: ${{ env.CHART_DIR }}
+
+      - name: Execute dry run
+        uses: Alfresco/alfresco-build-tools/.github/actions/helm-dry-run@AAE-10136-review-helm-actions
+        with:
+          chart-dir: ${{ env.CHART_DIR }}
+          test-rancher-url: ${{ secrets.RANCHER2_URL }}
+          test-rancher-access-key: ${{ secrets.RANCHER2_ACCESS_KEY }}
+          test-rancher-secret-key: ${{ secrets.RANCHER2_SECRET_KEY }}
+          test-cluster-name: ${{ env.TEST_CLUSTER_NAME }}
+          test-namespace: ${{ steps.define-namespace.outputs.name }}
+
+      - name: Package Helm Chart
+        id: package-helm-chart
+        uses: Alfresco/alfresco-build-tools/.github/actions/helm-package-chart@AAE-10136-review-helm-actions
+        with:
+          chart-dir: ${{ env.CHART_DIR }}
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ steps.package-helm-chart.outputs.package-file }}
+
+      - name: Publish Helm chart
+        uses: Alfresco/alfresco-build-tools/.github/actions/helm-publish-chart@AAE-10136-review-helm-actions
+        with:
+          chart-package: ${{ steps.package-helm-chart.outputs.package-file }}
+          helm-charts-repo: ${{ env.HELM_REPO }}
+          helm-charts-repo-branch: ${{ env.HELM_REPO_BRANCH }}
+          helm-charts-repo-subfolder: ${{ env.HELM_REPO_SUBFOLDER }}
+          helm-charts-repo-base-url: ${{ env.HELM_REPO_BASE_URL }}
+          token: ${{ secrets.BOT_GITHUB_TOKEN }}
+          git-username: ${{ secrets.BOT_GITHUB_USERNAME }}
+          do-push: ${{ github.event_name == 'push' }}
+
+  notify:
+    runs-on: ubuntu-latest
+    needs: build
+    if: always() && failure() && github.event_name == 'push'
```

Comment:
always() returns always true, failure() returns true only when a previous step failed, so always() is redundant here.

Reply:
I've had this conversation several times (for instance here: https://github.com/HylandSoftware/hxpr/pull/650#pullrequestreview-1037246728) and now that's the condition I put on all slack notification steps, because we want this to be executed always (including on cancellation)

Reply:
Honestly I'd like to avoid changing this in this PR, since I've been bitten by the issue several times, and this same expression is used on all slack notifs: here the notif is a dedicated job too, so I've seen failure use cases inside previous jobs where the notif was not issued unless you have "always()" in here (outside of cancellation).